crates/floresta-chain/src/pruned_utreexo/mod.rs

```diff
@@ -351,23 +355,119 @@ impl<T: BlockchainInterface> BlockchainInterface for Arc<T> {
     ) -> Result<Stump, Self::Error> {
         T::update_acc(self, acc, block, height, proof, del_hashes)
     }
-
     fn get_chain_tips(&self) -> Result<Vec<BlockHash>, Self::Error> {
         T::get_chain_tips(self)
     }
-
     fn validate_block(
         &self,
         block: &Block,
         proof: Proof,
-        inputs: HashMap<OutPoint, TxOut>,
+        inputs: UtxoMap,
         del_hashes: Vec<sha256::Hash>,
-        acc: Stump,
+        chain_time: &ChainTime,
     ) -> Result<(), Self::Error> {
-        T::validate_block(self, block, proof, inputs, del_hashes, acc)
+        T::validate_block(self, block, proof, inputs, del_hashes, chain_time)
     }
 
     fn get_fork_point(&self, block: BlockHash) -> Result<BlockHash, Self::Error> {
         T::get_fork_point(self, block)
     }
 }
+
+pub mod utxo_data {
+    use super::*;
+
+    pub struct UtxoData {
+        /// The transaction output that created this UTXO.
+        pub txout: bitcoin::TxOut,
+        //we need this mostly to validate relative-locktime.
+        /// The height of the block that the utxo was commited.
+        pub height: u64,
```

Comment:
Why is height a `u64`? We'll not have a block height greater than `U32_MAX` for the next 81715 years

Reply:
future proof